packages/connectivity/example/lib/main.dart

```diff
@@ -85,7 +86,49 @@ class _MyHomePageState extends State<MyHomePage> {
       appBar: AppBar(
         title: const Text('Plugin example app'),
       ),
-      body: Center(child: Text('Connection Status: $_connectionStatus\n')),
+      body: Center(child: Text(getText())),
```

Comment:
Instead of having this method here, and calling `getWifiName` and `getWifiIP` inside it, why don't you move the calls to `getWifiName` and `getWifiIP` to the `initConnectivity()` method?

Reply:
I don't think it's a good pattern to have these calls in the `build()` method - it's just an example, but I don't think they'll approve this PR with that.

I looked again your code in detail, what you think about that:

```dart

  @override
  void initState() {
    super.initState();
    initConnectivity();
    _connectivitySubscription =
        _connectivity.onConnectivityChanged.listen((ConnectivityResult result) {

      if (result == ConnectivityResult.wifi) { // <-- this is the change
        _checkWifi(result.toString());
      }
      else {
        setState(() => _connectionStatus = result.toString());
      }
    });
  }
```

And the `_checkWifi` would be:

```dart
  Future<void> _checkWifi(String wifiStatus) async {
    String wifiName, wifiIP;

    try {
      wifiName = (await _connectivity.getWifiName()).toString();
    } on PlatformException catch (e) {
      print(e.toString());
       wifiName = "Failed to get Wifi Name";
    }
     try {
      wifiIP = (await _connectivity.getWifiIP()).toString();
    } on PlatformException catch (e) {
      print(e.toString());
       wifiName = "Failed to get Wifi IP";
    }

    setState(() => 
      _connectionStatus = 'Connection Status: $wifiStatus\n'
      'Wifi Name: $wifiName\n'
      'Wifi IP: $wifiIP\n'  );      
  }
```

And the `build` would be have just the variable, without calls to the connectivity plugin.